### PR TITLE
[FEATURE] Add new registration list CSV hook

### PR DESCRIPTION
**Pull request by mk-mxp**
Includes the instance flushing from Tx_Seminars_Tests_Unit_BackEnd_RegistrationsListTest (required for the legacy unit tests).

### CODE REVIEW

File: Classes/Csv/AbstractRegistrationListView.php [hunk 1]
```diff
@@ -101,7 +103,15 @@ public function render()
...
         $allLines = array_merge([$this->createCsvHeading()], $this->createCsvBodyLines());
 
-        return $this->createCsvSeparatorLine() . implode(self::LINE_SEPARATOR, $allLines) . self::LINE_SEPARATOR;
+        $allLines = $this->createCsvSeparatorLine()
+            . implode(self::LINE_SEPARATOR, $allLines)
+            . self::LINE_SEPARATOR;
+
+        /** @var HookProvider $csvHookProvider */
+        $csvHookProvider = GeneralUtility::makeInstance(HookProvider::class, RegistrationListCsv::class);
+        $allLines = $csvHookProvider->executeHookReturningModifiedValue('modifyCsv', $allLines, $this);
```
Comment (oliverklee) [inline Classes/Csv/AbstractRegistrationListView.php:112]: `$allLines` now has two different meanings (all lines before and all lines after processing the hook(s)). Please either let's have two different varaiables, or directly return the hook results.

**Comment by mk-mxp** [comment]
Forgot that: this was the last hook on my list :) Thank you for accepting them!

**Comment by oliverklee** [comment]
You're welcome. Thanks for this massive list of contributions!

**Comment by mk-mxp** [comment]
Would you be willing to release these changes next week? I would very welcome using the latest release in an update at the beginning of June (instead of dev-master).

**Comment by oliverklee** [comment]
@mk-mxp I have just released version 3.2.0: https://github.com/oliverklee/ext-seminars/releases/tag/v3.2.0

**Comment by mk-mxp** [comment]
Thank you!

**Pull request merged**